### PR TITLE
initialize axios api call

### LINKED ISSUE #15: Install AXIOS & Request Data from APIs
Install AXIOS
Choice APIs
Create service (wrapper function) to alter the data retrieved from the API

### PR DESCRIPTION
**Pull request by Vince2C**
Create class wrapper for Axios

### CODE REVIEW

File: httpclientservice/httpClientService.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import axios from 'axios';
+
+class ClientService {
+  constructor() {
+    let service = axios.create({
+      headers: {}
```
Comment (mckchan13) [inline httpclientservice/httpClientService.js]: Should we import the API key from the .env as variables and place into the headers? 

```
//after the import axios from `axios` statement
const VACCOVID_API_KEY = process.env.VACCOVID_API_KEY

// then inside the axios.create instance:
let service = axios.create({
  headers: { 
   'x-rapidapi-host': 'vaccovid-coronavirus-vaccine-and-treatment-tracker.p.rapidapi.com',
   'x-rapidapi-key': 'VACCOVID_API_KEY',
   },
})
```

Comment (AndresGSystems) [reply httpclientservice/httpClientService.js]: No but we should be able to pass them as arguments on instantiation so the constructor can use them.

File: httpclientservice/httpClientService.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        break;
+    }
+    return Promise.reject(error)
+  }
+
+  redirectTo = (document, path) => {
+    document.location = path
+  }
+  
+  get(path, callback) {
```
Comment (AndresGSystems) [inline httpclientservice/httpClientService.js]: We just need the get method for the moment and the path should have a base url which in our API is common to any request and the string we want to append to that base url to complete the query.

File: httpclientservice/httpClientService.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import axios from 'axios';
+
+class ClientService {
```
Comment (AndresGSystems) [inline httpclientservice/httpClientService.js]: Nice job!! Just a couple of things,  names need to be improved a little bit. This class should be named something like HttpClientApiService or something around those lines.

File: httpclientservice/httpClientService.js (hunk 2 above)
Comment (AndresGSystems) [reply httpclientservice/httpClientService.js]: Let other methods to be implemented when needed. YAGNI pirnciple!!

**Review by Agfeynman — CHANGES_REQUESTED** [review]

File: httpclientservice/httpClientService.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  constructor() {
```
Comment (mckchan13) [inline httpclientservice/httpClientService.js]: Suggest modify the code as such:

```
const config = {
host: 'vaccovid-coronavirus-vaccine-and-treatment-tracker.p.rapidapi.com',
APIKey: 'a;lkjfds;al',
}

constructor( {host, APIKey} ) { 
  let service = axios.create({
  headers: { 
   'x-rapidapi-host': host,
   'x-rapidapi-key': APIKey,
   },
})}
```

**Comment by Vince2C** [comment]
Does it matter if it's in a class or not? Cause I don't think const can be used inside a class.

**Pull request merged**